featuretools/selection/selection.py

```diff
@@ -18,3 +21,120 @@ def remove_low_information_features(feature_matrix, features=None):
+            pct_corr_threshold (float): The correlation threshold to be considered highly correlated. Defaults to 0.95.
+            features_to_check (list[str], optional): List of column names to check whether any pairs are highly correlated.
+                        Will not check any other columns, meaning the only columns that can be removed are in this list.
+                        If null, defaults to checking all columns.
+            features_to_keep (list[str], optional): List of colum names to keep even if correlated to another column.
+                        If null, all columns will be candidates for removal.
+
+        Returns:
+            pd.DataFrame, list[:class:`.FeatureBase`]:
+                The feature matrix and the list of generated feature definitions. Matches dfs output.
+                If no feature list is provided as input, the feature list will not be returned.
+                For consistent results, do not change the order of features outputted by dfs.
+    """
+    if pct_corr_threshold < 0 or pct_corr_threshold > 1:
+        raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
+
+    if features_to_check is None:
+        features_to_check = feature_matrix.columns
+    else:
+        for f_name in features_to_check:
+            assert f_name in feature_matrix.columns, "feature named {} is not in feature matrix".format(f_name)
+
+    if features_to_keep is None:
+        features_to_keep = []
+
+    boolean = ['bool']
+    numeric_and_boolean_dtypes = vtypes.PandasTypes._pandas_numerics + boolean
+
+    fm_to_check = (feature_matrix[features_to_check]).select_dtypes(include=numeric_and_boolean_dtypes)
+
+    dropped = set()
+    columns_to_check = fm_to_check.columns
+    # When two features are found to be highly correlated, we drop the more complex feature
+    # Columns produced later in dfs are more complex
+    for i in range(len(columns_to_check) - 1, 0, -1):
+        more_complex_name = columns_to_check[i]
+        more_complex_col = fm_to_check[more_complex_name]
+
+        for j in range(i - 1, -1, -1):
+            less_complex_name = columns_to_check[j]
+            less_complex_col = fm_to_check[less_complex_name]
+
+            if abs(more_complex_col.corr(less_complex_col)) >= pct_corr_threshold:
+                dropped.add(more_complex_name)
+                break
+
+    keep = [f_name for f_name in feature_matrix.columns if (f_name in features_to_keep or
+                                                            f_name not in dropped)]
+    return _apply_feature_selection(keep, feature_matrix, features)
+
+
+def _apply_feature_selection(keep, feature_matrix, features=None):
+    new_matrix = feature_matrix[keep]
+    new_feature_names = set(new_matrix.columns)
+
+    if features is not None:
+        features = [f for f in features if f.get_name() in new_feature_names]
```

Comment:
have you tested this with a multi-output feature?  this might not handle the case where a feature has multiple columns associated with it

Reply:
I added a `test_multi_output_selection` and it's performing like I'd expect it to--not sure if the extra test is necessary though. Could maybe just view it as a multiple entity test as well

Reply:
`test_multioutput_selection` would need to pass in the feature list as well

Reply:
We can divide up the feature into FeatureOutputSlice sub-features if some but not all features should be kept

Reply:
Using `FeatureOutputSlice` to do this